### PR TITLE
chore: remove legacy event serialization

### LINKED ISSUE #12996: Remove legacy serialization
We do not need to support serialization of previous versions of events in the code any more. In order to continue the refactor, it is simpler to remove this code. This includes `BaseEventUnhashedData` which is no longer needed.

### PR DESCRIPTION
**Pull request by lpetrovic05**
closes #12996

### CODE REVIEW

**Review by anthony-swirldslabs — APPROVED** [review]
Platform Data changes look good to me.

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/event/validation/EventSignatureValidatorTests.java [hunk 1]
```diff
@@ -263,16 +251,19 @@ void ancientEvent(final boolean useBirthRoundForAncientThreshold) {
...
-        final BaseEventHashedData hData = event.getHashedData();
-        when(hData.getBirthRound()).thenReturn(EventConstants.MINIMUM_ROUND_CREATED);
-        when(hData.getGeneration()).thenReturn(EventConstants.FIRST_GENERATION);
-        when(event.getAncientIndicator(any())).thenAnswer(invocation -> {
-            final AncientMode mode = invocation.getArgument(0);
-            return mode == AncientMode.GENERATION_THRESHOLD
-                    ? EventConstants.FIRST_GENERATION
-                    : EventConstants.MINIMUM_ROUND_CREATED;
-        });
+        //        when(hData.getGeneration()).thenReturn(EventConstants.FIRST_GENERATION);
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/event/validation/EventSignatureValidatorTests.java:254]: commented out code block

**Review by edward-swirldslabs — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/GossipEvent.java [hunk 2]
```diff
@@ -50,9 +47,10 @@ private static final class ClassVersion {
         public static final int BIRTH_ROUND = 3;
     }
 
-    private int serializedVersion = ClassVersion.BIRTH_ROUND;
     private BaseEventHashedData hashedData;
-    private BaseEventUnhashedData unhashedData;
+    /** creator's signature for this event */
+    private byte[] signature;
```
[the remark this answers is not in the document]
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/GossipEvent.java:52]: This is coming in an upcoming PR, I have to add another feature to the `Bytes` class before I can do that

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/internal/EventImpl.java [hunk 3]
```diff
@@ -95,36 +93,12 @@ public class EventImpl extends EventMetadata
...
     }
 
-    public EventImpl(
-            final BaseEventHashedData baseEventHashedData,
-            final BaseEventUnhashedData baseEventUnhashedData,
-            final ConsensusData consensusData,
-            final EventImpl selfParent,
-            final EventImpl otherParent) {
-        this(new GossipEvent(baseEventHashedData, baseEventUnhashedData), consensusData, selfParent, otherParent);
+    public EventImpl(final GossipEvent gossipEvent) {
```
[the remark this answers is not in the document]
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/internal/EventImpl.java:100]: added

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/event/validation/EventSignatureValidatorTests.java (hunk 1 above)
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/event/validation/EventSignatureValidatorTests.java:254]: removed

**Review by alittley — APPROVED** [review]

**Review by anthony-swirldslabs — APPROVED** [review]

**Pull request merged**